index.html

```diff
@@ -4292,6 +4306,446 @@ <h3>Proof Formats</h3>
+verification functions, the <var>expectedProofPurpose</var> option is set to
+`assertionMethod` and if the <a>conforming document</a> is expected to be a
+<a>verifiable presentation</a>, the <var>domain</var> and <var>challenge</var>
+are also provided options.
+          </li>
+          <li>
+If <var>securingMechanismVerified</var>.<var>status</var> is set to
+`false`, add a <a href="#CRYPTOGRAPHIC_SECURITY_ERROR">
+  CRYPTOGRAPHIC_SECURITY_ERROR</a> to
+<var>securingMechanismVerified</var>.<var>errors</var>.
+          </li>
+          <li>
+If <var>securingMechanismVerified</var>.<var>status</var> is set to
+`true`:
+            <ol class="algorithm">
+              <li>
+Set <var>securingMechanismVerified</var>.<var>document</var> by passing
+<var>inputBytes</var> to the <var>extractDocument</var> function.
+              </li>
+              <li>
+Set <var>securingMechanismVerified</var>.<var>mediaType</var> to the media type
+associated with <var>securingMechanismVerified</var>.<var>document</var>.
+              </li>
+            </ol>
+          </li>
+          <li>
+Return <var>securingMechanismVerified</var>.
+          </li>
+        </ol>
+
+      </section>
+
+      <section>
+        <h3>Problem Details</h3>
+
+        <p>
+When an implementation detects an anomaly while processing a document, a
+<dfn>ProblemDetails</dfn> object can be used to report the issue to other
+software systems. The interface for these types of objects follows [[RFC9457]]
+to encode the data. A <a>ProblemDetails</a> object consists of the following
+properties:
+        </p>
+
+        <dl>
+          <dt>type</dt>
+          <dd>
+The `type` <a>property</a> MUST be present and its value MUST be a <a>URL</a>
+identifying the type of problem.
+          </dd>
+          <dt>code</dt>
+          <dd>
+The `code` <a>property</a> is an OPTIONAL
+<a data-cite="RFC9457#name-extension-members">RFC9457 extension member</a> . If
+present, its value MUST be an integer that identifies the type of the problem.
+Integer codes are useful in systems that only provide integer return values.
+          </dd>
+          <dt>title</dt>
+          <dd>
+The `title` <a>property</a> MUST be present and its value SHOULD provide a short
+but specific human-readable string for the problem.
+          </dd>
+          <dt>detail</dt>
+          <dd>
+The `detail` <a>property</a> MUST be present and its value SHOULD provide a
+longer human-readable string for the problem.
+          </dd>
+        </dl>
+
+        <p>
+Other properties that are useful for debugging purposes MAY be included in
+a <a>ProblemDetails</a> object. See [[RFC9457]] for further guidance on
+using this mechanism.
+        </p>
+
+        <p>
+The following problem description types and codes are defined by this
+specification:
+        </p>
+
+        <dl>
+          <dt id="PARSING_ERROR">
+https://www.w3.org/TR/vc-data-model#PARSING_ERROR
+(-64)
```

Comment:
There is a consistency issue between this approach at the VCDM and the DI spec (soon to be CR). 

In the DI case, the [error codes](https://www.w3.org/TR/vc-data-integrity/#processing-errors) are defined in the `https://w3id.org/security#` namespace, i.e., as part of the corresponding vocabulary; and these terms are all defined as part of the vocabulary ("individuals", in RDF parlance).

In the proposed case the terms are defined simply as part of the Document, with the official namespace being the VCDM document's short name. The terms are not defined as part of the [VCDM Vocabulary](https://www.w3.org/2018/credentials/) as RDF individuals, "just" as fragment URLs in the spec.

To be honest, I do not care too much which of the two mechanisms are used, but we should be consistent. The extra difficulty is that the DI document is currently (almost) at CR, so its modification _might_ happen as part of a new CR Snapshot (to be discussed separately).

Reply:
Yes, and I'm on the fence about where to define these things. Defining them in the vocabulary feels like overkill, but the nice thing about doing that is other specs could add to the vocabulary over time (the values don't need to only come from one place). The benefit of putting them in the spec is simplicity (no vocabulary to maintain over time).

I'm happy to go either way, we need stronger opinions on what to do here.

Reply:
I am also on the fence (as I was when we discussed this for the DI spec). The only thing that really matters, I believe, is consistency.

Reply:
> I'm happy to go either way, we need stronger opinions on what to do here.

My only argument is that the DI spec is in CR (since this morning), i.e., and such a change _might_ trigger a new CR snapshot (which is a more complex publishing process), whereas we are still pre-CR for the document we are discussing. But, as I said, this is not a really strong argument.